### PR TITLE
Migrating adaptive_app codelab to go_router

### PR DESCRIPTION
**Pull request by miquelbeltran**
In this PR, I am upgrading the `adaptive_app` code to use `go_router`, as part of the work on migrating all relevant codelabs to go_router.

The steps 04, 05, 06 and 07 have been upgraded. Step 03 remains the same.

Steps 4, 5, 6 are identical. The new GoRouter configuration has a default root `/`, and also can handle opening playlists providing an id, with `playlist/:id` using GoRouter params and queryparams to pass data.

In Step 7, there's also a login screen that will appear when the user is logged out. For that, it uses the `redirect` parameter to configure when the app should redirect to the login screen, and when it should navigate to the `/` root.

As a side note, the dependencies in the pubspec.yaml files have been sorted alphabetically.

FYI @domesticmouse @johnpryan @RedBrogdon 

## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I updated/added relevant documentation (doc comments with `///`).

Leaving this unmarked, as the Codelab document needs to be updated as well.

- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: adaptive_app/step_04/lib/main.dart [hunk 1]
```diff
@@ -13,6 +15,30 @@ const flutterDevAccountId = 'UCwXdFgeE9KYzlDdR7TG9cMw';
...
+          path: 'playlist/:id',
+          builder: (context, state) {
+            final title = state.queryParams['title']!;
+            final id = state.params['id']!;
+            return PlaylistDetails(
+              playlistId: id,
+              playlistName: title,
+            );
+          },
+        ),
```
Comment (miquelbeltran) [inline adaptive_app/step_04/lib/main.dart:36]: I am not 100% sure if the right way to do this is by having a sub-route like this, or instead having another "root" route as `/platlist/:id`. This implementation follows the go_router example.

File: adaptive_app/step_07/lib/src/adaptive_login.dart [hunk 2]
```diff
@@ -18,13 +22,13 @@ typedef _AdaptiveLoginButtonWidget = Widget Function({
 });
 
 class AdaptiveLogin extends StatelessWidget {
-  const AdaptiveLogin(
-      {required this.builder,
-      required this.clientId,
-      required this.scopes,
-      required this.loginButtonChild,
-      super.key});
-  final AdaptiveLoginBuilder builder;
```
Comment (miquelbeltran) [inline adaptive_app/step_07/lib/src/adaptive_login.dart:27]: The go_router implementation simplifies a bit the way the login status is handled, so an `AdaptiveLoginBuilder` is no longer required.

File: adaptive_app/step_04/lib/main.dart (hunk 1 above)
Comment (domesticmouse) [reply adaptive_app/step_04/lib/main.dart:36]: @johnpryan any thoughts? I have no opinion here.

**Review by domesticmouse — APPROVED** [review]
This LGTM, but I know nothing about `go_router`. Deferring to @johnpryan

**Comment by domesticmouse** [comment]
@miquelbeltran given that @johnpryan is busy as, I'm happy to land this as is. Let me know if this is ready to land...

**Comment by miquelbeltran** [comment]
> I'm happy to land this as is. Let me know if this is ready to land...

Got it! I will fix the conflicts and let you know

**Comment by miquelbeltran** [comment]
@domesticmouse main merge done, ready to land.

**Pull request merged**